File: src/client_gen/cpp/DebugInterface.cpp
#include "DebugInterface.hpp"
#include "model/ClientMessage.hpp"

DebugInterface::DebugInterface(const std::shared_ptr<OutputStream>& outputStream)
    : outputStream(outputStream)
{
}

void DebugInterface::send(const DebugCommand& command)
{
    // TODO: Construct actual message, this is a hack :)
    outputStream->write(ClientMessage::DebugMessage::TAG);
    command.writeTo(*outputStream);
    outputStream->flush();
}

File: src/client_gen/csharp/DebugInterface.cs
using System.IO;

namespace ProjectName
{
    public class DebugInterface
    {
        private BinaryWriter writer;
        public DebugInterface(BinaryWriter writer)
        {
            this.writer = writer;
        }
        public void Send(Model.DebugCommand command)
        {
            new Model.ClientMessage.DebugMessage(command).WriteTo(writer);
            writer.Flush();
        }
    }
}